include/heuristic/HeuristicMapper.hpp

```diff
@@ -60,98 +74,48 @@ class HeuristicMapper : public Mapper {
     Node() = default;
     Node(const std::array<std::int16_t, MAX_DEVICE_QUBITS>& q,
          const std::array<std::int16_t, MAX_DEVICE_QUBITS>& loc,
-         const std::vector<std::vector<Exchange>>&          sw = {}) {
+         const std::vector<std::vector<Exchange>>&          sw            = {},
+         const double                                       initCostFixed = 0)
+        : costFixed(initCostFixed) {
       std::copy(q.begin(), q.end(), qubits.begin());
       std::copy(loc.begin(), loc.end(), locations.begin());
       std::copy(sw.begin(), sw.end(), std::back_inserter(swaps));
     }
 
     /**
-     * @brief applies an in-place swap of 2 qubits in `qubits` and `locations`
-     * of the node
+     * @brief returns costFixed + costHeur + lookaheadPenalty
      */
-    void applySWAP(const Edge& swap, Architecture& arch) {
-      const auto q1 = qubits.at(swap.first);
-      const auto q2 = qubits.at(swap.second);
-
-      qubits.at(swap.first)  = q2;
-      qubits.at(swap.second) = q1;
-
-      if (q1 != -1) {
-        locations.at(static_cast<std::size_t>(q1)) =
-            static_cast<std::int16_t>(swap.second);
-      }
-      if (q2 != -1) {
-        locations.at(static_cast<std::size_t>(q2)) =
-            static_cast<std::int16_t>(swap.first);
-      }
+    [[nodiscard]] double getTotalCost() const {
+      return costFixed + costHeur + lookaheadPenalty;
+    }
 
-      if (arch.getCouplingMap().find(swap) != arch.getCouplingMap().end() ||
-          arch.getCouplingMap().find(Edge{swap.second, swap.first}) !=
-              arch.getCouplingMap().end()) {
-        swaps.back().emplace_back(swap.first, swap.second, qc::SWAP);
-      } else {
-        throw QMAPException("Something wrong in applySWAP.");
-      }
+    /**
+     * @brief returns costFixed + lookaheadPenalty
+     */
+    [[nodiscard]] double getTotalFixedCost() const {
+      return costFixed + lookaheadPenalty;
     }
 
+    /**
+     * @brief applies an in-place swap of 2 qubits in `qubits` and `locations`
+     * of the node
+     */
+    void applySWAP(const Edge& swap, Architecture& arch);
+
     /**
      * @brief applies an in-place teleportation of 2 qubits in `qubits` and
      * `locations` of the node
      */
-    void applyTeleportation(const Edge& swap, Architecture& arch) {
-      const auto q1 = qubits.at(swap.first);
-      const auto q2 = qubits.at(swap.second);
-
-      qubits.at(swap.first)  = q2;
-      qubits.at(swap.second) = q1;
+    void applyTeleportation(const Edge& swap, Architecture& arch);
```

Comment:
Shouldn't it be possible for `arch` to be a `const` reference?

Reply:
Well, I just copied that function from `hpp` to `cpp`. And it seems, that it is not a `const` reference for a reason, since `arch.getTeleportationQubits()` (which is called in `applyTeleportation`) is not a `const` function and throws an error, when it turned into one.

Reply:
Right... thought that is was missing for a reason. In that case, nothing to do here 🤷🏽‍♂️